lib/compat/wordpress-6.5/fonts/class-wp-rest-font-faces-controller.php

```diff
@@ -858,7 +858,21 @@ protected function sanitize_src( $value ) {
 		 */
 		protected function handle_font_file_upload( $file ) {
 			add_filter( 'upload_mimes', array( 'WP_Font_Utils', 'get_allowed_font_mime_types' ) );
-			add_filter( 'upload_dir', 'wp_get_font_dir' );
+
+			/*
+			 * Set the upload directory to the fonts directory.
+			 *
+			 * wp_get_font_dir() contains the 'font_dir' hook, whose callbacks are
+			 * likely to call wp_get_upload_dir().
+			 *
+			 * To avoid an infinite loop, don't hook wp_get_font_dir() to 'upload_dir'.
+			 * Instead, just pass its return value to the 'upload_dir' callback.
+			*/
```

Comment:
```suggestion
			 * To avoid an infinite loop, don't hook wp_get_font_dir() to 'upload_dir'.
			 * Instead, just pass its return value to the 'upload_dir' callback.
			 */
```
EVIL IDEs